_plugins/code_parser.rb

```diff
@@ -0,0 +1,20 @@
+# Description : This plugin checks code blocks created using ``` for ASCCI characters that can cause problem pasted to terminal.
+# Authour : Reza R <54559947+frozenprocess@users.noreply.github.com>
+#####
+Jekyll::Hooks.register :pages, :post_render do |post|
+    # Check content of the post for code blocks.
+    code_blocks = post.content.scan(/<code.*>(.*?)<\/code>/m)
+    # Check code blocks for characters and saving their position in order to generate log.
+    occurences = code_blocks.join(",").gsub(/[–—‘’‚“”„…]/).map { Regexp.last_match.begin(0) }
```

Comment:
I haven't touched Ruby since 1.8 (so forgive me a little as it's been well over a decade 😄).

I think we may be able to do something a little more clever here than matching the existing known problematic utf-8 characters. With Ruby 1.9 it looks like string encoding was added ( https://ruby-doc.org/core-1.9.3/String.html#method-i-encode ). What if we did  `Encoding.find('ASCII')` for the type, and then on conversion failure print the offending character in the callback. This way if the character can be converted, they will (or can) be, and if it simply cannot be converted (multiple failure modes), user intervention is required.

What are your thoughts on this one?

Reply:
Same here, I did use Ruby while ago and its like learning it again from scratch now 😄 .

Initially I proposed something similar. however, after talking to @lxpollitt we decided instead of converting characters break the compiling cycle and ask user to intervene .